cloud/filestore/libs/storage/service/service_ut_sharding.cpp

```diff
@@ -4019,6 +4019,133 @@ Y_UNIT_TEST_SUITE(TStorageServiceShardingTest)
             true    // reboot tablet
         );
     }
+
+    Y_UNIT_TEST(ShouldBalanceShardsByFreeSpace)
+    {
+        NProto::TStorageConfig config;
+        config.SetShardIdSelectionInLeaderEnabled(true);
+        config.SetAutomaticShardCreationEnabled(true);
+        config.SetShardAllocationUnit(4_MB);
+        config.SetAutomaticallyCreatedShardSize(4_MB);
+        config.SetShardBalancerMinFreeSpaceReserve(4_KB);
+        config.SetShardBalancerDesiredFreeSpaceReserve(1_MB);
+        config.SetMultiTabletForwardingEnabled(true);
+
+        TTestEnv env({}, config);
+        env.CreateSubDomain("nfs");
+
+        ui32 nodeIdx = env.CreateNode("nfs");
+
+        const TString fsId = "test";
+
+        TServiceClient service(env.GetRuntime(), nodeIdx);
+        service.CreateFileStore(fsId, 20_MB / 4_KB);
+
+        // waiting for IndexTablet start after the restart triggered by
+        // configureshards
+        WaitForTabletStart(service);
+
+        TVector<TString> expected = {
+            fsId,
+            fsId + "_s1",
+            fsId + "_s2",
+            fsId + "_s3",
+            fsId + "_s4",
+            fsId + "_s5",
+        };
+        auto listing = service.ListFileStores();
+        auto fsIds = listing->Record.GetFileStores();
+        TVector<TString> ids(fsIds.begin(), fsIds.end());
+        Sort(ids);
+        UNIT_ASSERT_VALUES_EQUAL(expected, ids);
+
+        auto headers = service.InitSession(fsId, "client");
+
+        TVector<ui64> handles;
+        TVector<ui64> nodes;
+        TSet<ui32> shards;
+        for (ui32 i = 0; i < 5; ++i) {
+            auto createHandleResponse = service.CreateHandle(
+                headers,
+                fsId,
+                RootNodeId,
+                Sprintf("file%u", i),
+                TCreateHandleArgs::CREATE)->Record;
+
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
+
+            const auto handleId = createHandleResponse.GetHandle();
+            handles.push_back(handleId);
+        }
+
+        UNIT_ASSERT_VALUES_EQUAL(5, shards.size());
+
```

Comment:
let's add some comments to the test to make it more understandable
such as 
// filled 3 out of 5 shards

Reply:
will be resolved in next pr